File: huawei-push-kit/src/test/scala/docs/scaladsl/PushKitExamples.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * license agreements; and to You under the Apache License, version 2.0:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * This file is part of the Apache Pekko project, which was derived from Akka.
 */

/*
 * Copyright (C) since 2016 Lightbend Inc. <https://www.lightbend.com>
 */

package docs.scaladsl

import org.apache.pekko.actor.ActorSystem
//#imports
import org.apache.pekko
import pekko.stream.connectors.huawei.pushkit._
import pekko.stream.connectors.huawei.pushkit.scaladsl.HmsPushKit
import pekko.stream.connectors.huawei.pushkit.models.AndroidConfig
import pekko.stream.connectors.huawei.pushkit.models.AndroidNotification
import pekko.stream.connectors.huawei.pushkit.models.BasicNotification
import pekko.stream.connectors.huawei.pushkit.models.ClickAction
import pekko.stream.connectors.huawei.pushkit.models.Condition
import pekko.stream.connectors.huawei.pushkit.models.ErrorResponse
import pekko.stream.connectors.huawei.pushkit.models.PushKitNotification
import pekko.stream.connectors.huawei.pushkit.models.PushKitResponse
import pekko.stream.connectors.huawei.pushkit.models.Response
import pekko.stream.connectors.huawei.pushkit.models.Tokens

//#imports
import pekko.stream.scaladsl.Source
import pekko.stream.scaladsl.Sink

import scala.collection.immutable
import scala.concurrent.Future

class PushKitExamples {

  implicit val system = ActorSystem()

  // #simple-send
  val config = HmsSettings()
  val notification: PushKitNotification =
    PushKitNotification.empty
      .withNotification(
        BasicNotification.empty
          .withTitle("title")
          .withBody("body"))
      .withAndroidConfig(
        AndroidConfig.empty
          .withNotification(
            AndroidNotification.empty
              .withClickAction(
                ClickAction.empty
                  .withType(3))))
      .withTarget(Tokens(Set[String]("token").toSeq))

  Source
    .single(notification)
    .runWith(HmsPushKit.fireAndForget(config))
  // #simple-send

  // #asFlow-send
  val result1: Future[immutable.Seq[Response]] =
    Source
      .single(notification)
      .via(HmsPushKit.send(config))
      .map {
        case res @ PushKitResponse(code, msg, requestId) =>
          println(s"Response $res")
          res
        case res @ ErrorResponse(errorMessage) =>
          println(s"Send error $res")
          res
      }
      .runWith(Sink.seq)
  // #asFlow-send

  // #condition-builder
  import org.apache.pekko.stream.connectors.huawei.pushkit.models.Condition.{ Topic => CTopic }
  val condition = Condition(CTopic("TopicA") && (CTopic("TopicB") || (CTopic("TopicC") && !CTopic("TopicD"))))
  // #condition-builder
}

File: huawei-push-kit/src/test/java/docs/javadsl/PushKitExamples.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * license agreements; and to You under the Apache License, version 2.0:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * This file is part of the Apache Pekko project, derived from Akka.
 */

/*
 * Copyright (C) since 2016 Lightbend Inc. <https://www.lightbend.com>
 */

package docs.javadsl;

import org.apache.pekko.actor.ActorSystem;

// #imports
import org.apache.pekko.stream.connectors.huawei.pushkit.*;
import org.apache.pekko.stream.connectors.huawei.pushkit.javadsl.HmsPushKit;
import org.apache.pekko.stream.connectors.huawei.pushkit.models.AndroidConfig;
import org.apache.pekko.stream.connectors.huawei.pushkit.models.AndroidNotification;
import org.apache.pekko.stream.connectors.huawei.pushkit.models.BasicNotification;
import org.apache.pekko.stream.connectors.huawei.pushkit.models.ClickAction;
import org.apache.pekko.stream.connectors.huawei.pushkit.models.ErrorResponse;
import org.apache.pekko.stream.connectors.huawei.pushkit.models.PushKitNotification;
import org.apache.pekko.stream.connectors.huawei.pushkit.models.PushKitResponse;
import org.apache.pekko.stream.connectors.huawei.pushkit.models.Response;
import org.apache.pekko.stream.connectors.huawei.pushkit.models.Tokens;

// #imports
import org.apache.pekko.stream.javadsl.Sink;
import org.apache.pekko.stream.javadsl.Source;
import scala.collection.immutable.Set;

import java.util.List;
import java.util.concurrent.CompletionStage;

public class PushKitExamples {

  public static void example() {
    ActorSystem system = ActorSystem.create();

    // #simple-send
    HmsSettings config = HmsSettings.create(system);
    PushKitNotification notification =
        PushKitNotification.fromJava()
            .withNotification(BasicNotification.fromJava().withTitle("title").withBody("body"))
            .withAndroidConfig(
                AndroidConfig.fromJava()
                    .withNotification(
                        AndroidNotification.fromJava()
                            .withClickAction(ClickAction.fromJava().withType(3))))
            .withTarget(new Tokens(new Set.Set1<>("token").toSeq()));

    Source.single(notification).runWith(HmsPushKit.fireAndForget(config), system);
    // #simple-send

    // #asFlow-send
    CompletionStage<List<Response>> result =
        Source.single(notification)
            .via(HmsPushKit.send(config))
            .map(
                res -> {
                  if (!res.isFailure()) {
                    PushKitResponse response = (PushKitResponse) res;
                    System.out.println("Response " + response);
                  } else {
                    ErrorResponse response = (ErrorResponse) res;
                    System.out.println("Send error " + response);
                  }
                  return res;
                })
            .runWith(Sink.seq(), system);
    // #asFlow-send
  }
}
